### PR TITLE
Argo server should not fail if configmap name is not provided or config map does not exist

### PR DESCRIPTION
**Pull request by alexmt**

### CODE REVIEW

File: cmd/argocd-server/commands/root.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func NewCommand() *cobra.Command {
 	command.Flags().StringVar(&staticAssetsDir, "staticassets", "", "Static assets directory path")
 	command.Flags().StringVar(&logLevel, "loglevel", "info", "Set the logging level. One of: debug|info|warn|error")
 	command.Flags().StringVar(&repoServerAddress, "repo-server", "localhost:8081", "Repo server address.")
-	command.Flags().StringVar(&configMapName, "config-map", "", "Name of a Kubernetes config map to use.")
+	command.Flags().StringVar(&configMapName, "config-map", "argo-cd-cm", "Name of a Kubernetes config map to use.")
```
Comment (merenbach) [inline cmd/argocd-server/commands/root.go]: Might it make sense to replace `"argo-cd-cm"` with a `util.ConfigManagerDefaultConfigMapName` const?  That way the same default const can be used between `argocd install` and `argocd-server`.

**Review by merenbach — APPROVED** [review]
Looks good overall.  Please see comment regarding magic number (well, string).

File: cmd/argocd-server/commands/root.go (hunk 1 above)
Comment (alexmt) [reply cmd/argocd-server/commands/root.go]: @merenbach for some reason I cannot find ConfigManagerDefaultConfigMapName const. Is it in master or some other branch?

Comment (merenbach) [reply cmd/argocd-server/commands/root.go]: @alexmt I should have clarified.  This would be a new const in `util/configmanager.go`.

Comment (alexmt) [reply cmd/argocd-server/commands/root.go]: Got it. It make sense, working on changes

**Comment by alexmt** [comment]
@merenbach please take another look. I had to make more changes in intall.go to handle situation when config-map name is not provided by user, but kubernetes has config map with default name exist and has non-default secret name.

**Comment by merenbach** [comment]
@alexmt looks good to me!

**Pull request merged**